packages/nimble-components/src/utilities/unit-format/decimal-unit-format.ts

```diff
@@ -18,7 +18,6 @@ export class DecimalUnitFormat extends UnitFormat<DecimalUnitFormatOptions> {
     private readonly maximumFractionDigits: number;
```

Comment:
We had a similar kind of issue with the dialog api, so maybe we can have a shared file of DOM type definition updates we can burn down over time: https://github.com/ni/nimble/blob/main/packages/nimble-components/src/dialog/index.ts#L25

Reply:
I don't think it's possible to modify the type of an existing property on an interface (which is what I need to make `signDisplay` accept a value of `'negative'`). I was able to use declaration merging to add the missing `roundingPriority` property, though. Seems like I can put the .d.ts file anywhere under `/src` and it gets pulled in automatically. I created `type-extensions.d.ts` directly under `/utilities` and put both the extended `Intl.NumberFormatOptions` and `HTMLDialogElement` types in there. `Dialog` and `Drawer` each had a `dialog` property that I changed from being `ExtendedDialog` to `HTMLDialogElement`, but they should be internal, so no breaking change.

Reply:
> Seems like I can put the .d.ts file anywhere under /src and it gets pulled in automatically.

Can you provide more evidence about why that is working. It's not clear to me what is making that association. I'd expect at a minimum an import or type reference to the file.

> I don't think it's possible to modify the type of an existing property on an interface

I'm pretty sure that is possible. You might need to make sure you are modifying the correct type through the global

Reply:
> Can you provide more evidence about why that is working.

Our tsconfig sets `include` to `["src"]`. This will [make Typescript aware](https://www.typescriptlang.org/tsconfig#include) of any *.d.ts files under `src`. Because we are extending built-in types (which do not require explicit imports), our extensions are automatically used without explicit imports. I can't find direct documentation of this, but [this StackOverflow answer](https://stackoverflow.com/questions/58265985/extending-built-in-types-in-typescript) implies it, because the .d.ts file never has to be imported anywhere.

> I'm pretty sure that is possible.

I've spent more time than I'd like trying and failing. The [best answer StackOverflow has](https://stackoverflow.com/questions/41285211/overriding-interface-property-type-defined-in-typescript-d-ts-file) is to create a new interface that modifies the property type of an existing interface. I can't find a way to change the existing interface (other than adding to it).

Reply:
Could you test with a npm pack tgz build in an app that does not enable skipLibCheck (so we know it's type checking us). If that works then I'm sold. 💯

Reply:
I updated TestInsights's dependency to use a tgz of `nimble-components`. Then I added some code to create and use a `NumberTextUnitFormat` (to make sure it uses the code where we needed the extended `Intl.NumberFormatOptions` definition). It built without issue. Is that sufficient? The app build won't be type-checking the js logic in `nimble-components`, and none of our public APIs include `Intl.NumberFormatOptions` or `HTMLDialogElement`, so I'm not really sure what I need to verify.

Reply:
We are modifying a global type that's not scoped to our components or apis. I'd guess our types are included in the build. By importing our lib and having a dep on it I'd expect the global types in that app to be mutated to have the properties we define. So without components dialog, etc would not have the type and with it it should. So HTMLDialogElement etc should be changed for those apps after importing our lib. And my concern about aligning the types to prevent conflicts.

Reply:
From testing, it seems our type extensions are not being included by the client app. I don't think that's necessarily a problem, though. As long as the extended types are not part of our public API (e.g. we don't expect clients to pass us `Intl.NumberFormatOptions`), isn't it actually better that we don't modify the client's type definitions? As long as our builds work and client builds work, haven't our type extensions served their purpose?

Reply:
After a chat with @rajsite I was confused about how this was working so I did a bit of reading / thinking. I think what's happening is:
1. nimble-components consumes the custom .d.ts file during its build to do type checking on anything that uses those types
     - I didn't track down exactly how `tsc` finds those .d.ts files but I suspect it's because the tsconfig.json says `include: "src"`
2. the build output of nimble-components is just js files and .d.ts files for the public APIs of those JS files
     - ok, technically the build output also contains source maps which point to inlined ts in the .d.ts files, but that's just used for runtime debugging, not app compilation
3. the build output of nimble-components does NOT include the custom internal .d.ts file
4. when an app builds against nimble-components it doesn't try to do any type checking of the contents of nimble. skipLibCheck just controls whether it type checks the .d.ts files of libraries
5. As long as Nimble doesn't use any of the augmented types in its public API, they won't impact the app's build at all.

So based on my understanding, I think the current approach is sane.

Reply:
That was my understanding, too. 👍

Reply:
Gotcha, it was not clear to me that the .d.ts files are not included in the build output at all.

>  use any of the augmented types in its public API 

> As long as the extended types are not part of our public API 

HTMLDialogElement is in the public api from TypeScript's perspective:
![image](https://github.com/ni/nimble/assets/1588923/8dd2eb36-0ddd-4e48-84a3-e5b3344d961a)

Same for Intl.NumberFormatOptions:
![image](https://github.com/ni/nimble/assets/1588923/f1bfa994-1fe0-46ae-ba2b-aafdb315e6dc)

But I agree that `showModal`, `close` ,`signDisplay`, and `roundingPriority` are not.

Is that what y'all are saying? How did y'all validate that? The same way I did by text searching all the .d.ts files?

I think we should capture some docs on this for future editors of the .d.ts files. @m-akinc I'll resolve this comment after adding additional ones.

Reply:
> But I agree that showModal, close ,signDisplay, and roundingPriority are not.
> 
> Is that what y'all are saying? How did y'all validate that? The same way I did by text searching all the .d.ts files?

I added a comment about how I confirmed that these types are not in the public api. If y'all have another approach that was used can mention there.

Reply:
> Is that what y'all are saying?

I was saying that our APIs do not expect/require clients to use `showModal`, `close`, `signDisplay`, or `roundingPriority`, so we shouldn't care about whether the client's definitions of `HTMLDialogElement` or `Intl.NumberFormatOptions` contain them.

> How did y'all validate that?

To see if the client's definitions had those functions/properties, I added some code to TestInsights that tried to use `showModal()` on `HTMLDialogElement` and set `roundingPriority` on `Intl.NumberFormatOptions`. The build (and editor) gave me errors about `roundingPriority`. Interestingly, though, it did not complain about `showModal()`, because it seemed to be using a definition from a typescript installation in the VS Code app data directory.